PageSpeed_Page_View_FromAPI.php

```diff
@@ -0,0 +1,816 @@
+		<div class="percentage"></div>
+		<div class="mask"></div>
+		<span class="value">
+			<i class="material-icons" aria-hidden="true"><?php echo esc_html( $icon ); ?></i>
+			<?php echo ( ! empty( $data['score'] ) ? esc_html( $data['score'] ) : '' ); ?>
+		</span>
+	</div>
+	<?php
+}
+
+/**
+ * Get PageSpeed metric notice BG
+ *
+ * @param int $score PageSpeed desktop/mobile score.
+ *
+ * @return string
+ */
+function w3tcps_breakdown_bg( $score ) {
+	$notice = 'notice notice-info inline';
+	if ( ! empty( $score ) && is_numeric( $score ) ) {
+		if ( $score >= 90 ) {
+			$notice = 'notice notice-success inline';
+		} elseif ( $score >= 50 && $score < 90 ) {
+			$noitce = 'notice notice-warning inline';
+		} elseif ( $score > 0 && $score < 50 ) {
+			$notice = 'notice notice-error inline';
+		}
+	}
+	return $notice;
+}
+
+/**
+ * Get PageSpeed metric grade
+ *
+ * @param int $score PageSpeed desktop/mobile score.
+ *
+ * @return string
+ */
+function w3tcps_breakdown_grade( $score ) {
+	$grade = 'w3tcps_blank';
+	if ( ! empty( $score ) && is_numeric( $score ) ) {
+		if ( $score >= 90 ) {
+			$grade = 'w3tcps_pass';
+		} elseif ( $score >= 50 && $score < 90 ) {
+			$grade = 'w3tcps_average';
+		} elseif ( $score > 0 && $score < 50 ) {
+			$grade = 'w3tcps_fail';
+		}
+	}
+	return $grade;
+}
+
+/**
+ * Render the final generated screenshot
+ *
+ * @param array $data PageSpeed data.
+ *
+ * @return void
+ */
+function w3tcps_final_screenshot( $data ) {
+	if ( isset( $data ) && ! empty ( $data['screenshots']['final']['screenshot'] ) ) {
+		echo '<img src="' . esc_attr( $data['screenshots']['final']['screenshot'] ) . '" alt="' . ( ! empty ( $data['screenshots']['final']['title'] ) ? esc_attr( $data['screenshots']['final']['title'] ) : esc_attr__( 'Final Screenshot', 'w3-total-cache' ) ) . '"/>';
+	}
+}
+
+/**
+ * Render all "building" screenshots
+ *
+ * @param mixed $data PageSpeed desktop/mobile score.
+ *
+ * @return void
+ */
+function w3tcps_screenshots( $data ) {
+	if ( isset( $data ) && ! empty ( $data['screenshots']['other']['screenshots'] ) ) {
+		foreach ( $data['screenshots']['other']['screenshots'] as $screenshot ) {
+			echo '<img src="' . esc_attr( $screenshot['data'] ) . '" alt="' . ( ! empty ( $data['screenshots']['other']['title'] ) ? esc_attr( $data['screenshots']['other']['title'] ) : esc_attr__( 'Other Screenshot', 'w3-total-cache' ) ) . '"/>';
+		}
+	}
+}
+
+/**
+ * Render all metric data into listable items
+ *
+ * @param array $data PageSpeed desktop/mobile score.
+ *
+ * @return void
+ */
+function w3tcps_breakdown( $data ) {
+	if ( ! isset( $data ) || ( empty ( $data['opportunities'] ) && empty ( $data['diagnostics'] ) ) ) {
+		return;
+	}
+
+	$opportunities = '';
+	$diagnostics   = '';
+	$passed_audits = '';
+
+	foreach ( $data['opportunities'] as $opportunity ) {
+		if ( empty( $opportunity['details'] ) ) {
```

Comment:
I can see the "what", but not the why. In what scenario will we not have details? (please add a comment to the code)